### PR TITLE
Document versioning scheme

### PR DESCRIPTION
**Pull request by tautschnig**
This is based on the summary in
https://github.com/diffblue/cbmc/issues/5453#issuecomment-681887846 and
documents our current state of affairs.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- n/a Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: doc/ADR/release_process.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release Process
 
-**Date**: 8-10-2020
+**Date**: 2022-06-23
```
Comment (NlightNFotis) [inline doc/ADR/release_process.md]: Does it make sense to keep the old date and add a `Updated` field?

File: doc/ADR/release_process.md [hunk 2]
```diff
@@ -26,19 +26,41 @@ anything more, but the process is described below for reference:
...
+## Versioning
+
+We adopt an approach approximating SemVer. That is, our version numbers should
+be major.minor.patch. Regular releases (as of 2022-06-23: every other week)
+should normally increment the minor version number. This is also where we can
+deprecate undesired features, i.e., mark as deprecated, warn existing users.
+
+Patch version increments should not be done as part of the regular releases,
+and instead should be reserved for critical bugfixes. Don't make a patch
+release if you're not confident your release contains nothing but bugfixes.
```
Comment (NlightNFotis) [inline doc/ADR/release_process.md]: We sometimes do patch releases when we fix any problems with our CI infrastructure . Does it make sense to mention that?

**Review by NlightNFotis — APPROVED** [review]
LGTM, minor questions.

File: doc/ADR/release_process.md (hunk 1 above)
Comment (tautschnig) [reply doc/ADR/release_process.md]: Done.

File: doc/ADR/release_process.md (hunk 2 above)
Comment (tautschnig) [reply doc/ADR/release_process.md]: Done.

**Review by thomasspriggs — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**